opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/aggregation.py

```diff
@@ -370,7 +370,7 @@ def collect(
 
         return NumberDataPoint(
             attributes=self._attributes,
-            start_time_unix_nano=0,
+            start_time_unix_nano=None,
```

Comment:
I think even with that fix (which is fine for me), we'll still get `StartTimestamp: 1970-01-01 00:00:00 +0000 UTC`. It seems that even if we pass `None` during the encode [here](https://github.com/open-telemetry/opentelemetry-python/blob/main/exporter/opentelemetry-exporter-otlp-proto-common/src/opentelemetry/exporter/otlp/proto/common/_internal/metrics_encoder/__init__.py#L214), in the end, we'll have the timestamp as 0 on the collector 🤔. I just ran the example in the issue using this branch and still see the zeroed starttimestamp.

Reply:
Does this fix address the actual issue then?

Reply:
Nope, from the link above it looks like we are not serializing the field

Reply:
It makes no real-world difference to set `start_time_unix_nano` to `0` or `None`. We still should either set it to `None` or to another positive, non-zero value:

[This `pt` object](https://github.com/open-telemetry/opentelemetry-python/blob/main/exporter/opentelemetry-exporter-otlp-proto-common/src/opentelemetry/exporter/otlp/proto/common/_internal/metrics_encoder/__init__.py#L214) will have its `start_time_unix_nano` attribute set to `0` if we set [`NumberDataPoint.start_time_unix_nano`](https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/point.py#L35) to `0` or `None`. Since the definition of the `pt` object is outside of our repos, there is nothing we can do about its behavior. If we set `NumberDataPoint.start_time_unix_nano` to a non-zero positive value `x`, then `pt.start_time_unix_nano == x`.

I find 2 contradictory indications on what to do with this attribute:

1. [`NumberDataPoint.start_time_unix_nano`](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/metrics/v1/metrics.proto#L210) is ignored.
2. [`NumberDataPoint.start_time_unix_nano`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/data-model.md#gauge) is optional and can be set to a particular time when the "collection system started" (I don't actually know when that happens).

If we set it to `None` the exported result is the same as setting it to `0`. Regardless of that, we are doing the best that we can by setting it to `None` to indicate that such value is meaningless and should be ignored. If something else then changes it to `0`, well that is the problem of that something else.

The right thing to do would be to have a `GaugeDataPoint`, equal to `NumberDataPoint` but without the `start_time_unix_nano` attribute, achieving such thing is probably hard since it means changing the protos.

We could set it to the time when the "collection system started", but first we need to figure out what is the "collection system" and when did it "start". Because of this lack of definition, I say we do what we can and set `NumberDataPoint.start_time_unix_nano` for gauges to `None`.